src/components/Book.js

```diff
@@ -1,15 +1,15 @@
 import PropTypes from 'prop-types';
 import { useDispatch } from 'react-redux';
-import { removeBook } from '../redux/features/books/booksSlice';
+import { deleteBook } from '../redux/features/books/booksSlice';
 
 const Book = ({ bookInfo }) => {
   const dispatch = useDispatch();
   return (
-    <li>
+    <li key={bookInfo.item_id}>
       <div className="book-content">
         <p>{bookInfo.title}</p>
         <span>{bookInfo.author}</span>
-        <button type="button" onClick={() => dispatch(removeBook(bookInfo.item_id))}>Remove</button>
+        <button type="button" onClick={() => dispatch(deleteBook(bookInfo.item_id))}>Remove</button>
       </div>
     </li>
   );
```

Comment:
- [ ] Your logic is on point but there is something wrong with the rendering. sometimes the new books `li` are added to the book list `ul` and sometimes the author and title don't appear. only the remove button appears. and if you add 3 or 4 books like that and then click on any remove button, then all the `li`s gets removed. you can see this in the following video. kindly consider fixing this bug.

https://user-images.githubusercontent.com/96683843/230124691-864f0e3b-2974-4892-b1d7-4538dc5d4cf2.mov

Reply:
Thanks so much for your insights @Zuhaib042, I've fixed the bug. 🙏